### ISSUE TITLE
Recommended workflow to integrate onavg in a neuroimaging pipeline

### ISSUE THREAD

**Issue by sina-mansour** [feilong/tpl-onavg#1]
Hi,

First of all, thanks so much for openly providing these standard onavg surface templates for future use.

Checking the files on the repository, I wasn't able to find a clear explanation of how this resource can be used in future work. Namely, assuming I have already performed FreesSurfer processing (i.e. recon-all) and am interested to use the onavg surface for my study, how can I transform/register maps to the onavg? Are there any resources (e.g. tutorials/documentation) that details this?

Many thanks,
Sina

**Comment by feilong** [comment]
Hi Sina,

Thank you for your interest in our work. Here's the tutorial of using onavg with FreeSurfer: https://feilong.github.io/tpl-onavg/how_to_use/how_to_use_w_freesurfer.html

If you run the `recon-all` through fMRIPrep, you might also be interested in the code we wrote to resample to onavg based on fMRIPrep output: https://feilong.github.io/tpl-onavg/how_to_use/how_to_use_w_python.html

Best,
Feilong

**Comment by rubenSaro** [comment]
Hi,

Is it possible to use onavg in the HCP pipeline, instead of using the templates they use now.

Thanks

**Comment by feilong** [comment]
Yes.  If your data is registered to fsLR using MSM/HCPpipelines, you can use the `space-fsLR` files for resampling.
See https://github.com/templateflow/tpl-onavg

**Comment by rubenSaro** [comment]
Ok, so I can do the resampling of the surface BOLD timeseries after my full.
I see in the example (https://feilong.github.io/tpl-onavg/how_to_use/space_travel.html) it is used to resample some correlation data. It can be used also for BOLD timeseries?
HCP pipeline?

**Comment by feilong** [comment]
I recommend changing the last resampling step so that you resample directly to onavg, using the `space-fsLR` files.
Each time you resample your data, you loss some information.  If you resample again after the full pipeline (resampling to fsLR, then fsLR to onavg), you loss more information than resampling directly to onavg.

**Comment by rubenSaro** [comment]
Ok. So the strategy would be to adapt the HCP pipeline so it resamples to onavg instead to the freesurfer templates? Probably in here: GenericfMRISurfaceProcessingPipeline.sh
Thanks
